src/storage/disk_manager.c

```diff
@@ -5917,19 +5880,28 @@ disk_dump_goodvol_all (THREAD_ENTRY * thread_p, INT16 volid, void *arg)
 STATIC_INLINE bool
 disk_is_valid_volid (VOLID volid)
 {
-  // a passive transaction server (which is, implicitly, a transaction server
-  // with remote storage) maintains a separate bookkeeping of permanent data volumes
-  // which is updated when executing corresponding recovery replication functions
-  // as part of the regular replication; this way, it is able to provide a valid
-  // answer via this function
-  const bool is_valid_perm_volid = is_tran_server_with_remote_storage ()?
-    (volid < disk_Page_server_perm_volume_count) : (volid < disk_Cache->nvols_perm);
+  if (is_tran_server_with_remote_storage () && is_passive_transaction_server ())
+    {
+      /* Since PTS do not have a disk header page for newly added volume at the time of replication,
+       * disk_rv_redo_format (), which requires disk header page and does update the disk_Cache,
+       * is not called during replication.Thereefore, PTS knows only temporary volumes.
+       * Only the temporary volume ID is checked, and if it is not a temporary volume, only the range of volume id is checked */
+
+      const bool is_valid_temp_volid = (volid > (LOG_MAX_DBVOLID - disk_Cache->nvols_temp));
+      if (is_valid_temp_volid)
+	{
+	  return is_valid_temp_volid;
+	}
+
+      return volid >= LOG_DBFIRST_VOLID && volid <= LOG_MAX_DBVOLID;
+    }
+
+  const bool is_valid_perm_volid = (volid < disk_Cache->nvols_perm);
```

Comment:
Maybe make this check the same as in develop since it serves the "old" purpose - for the "monolithic" server and will make back-merging easier.
```suggestion
    return volid < disk_Cache->nvols_perm || volid > LOG_MAX_DBVOLID - disk_Cache->nvols_temp;
```

Reply:
OK, Thanks. applied